lib/fika/parser.ex

```diff
@@ -254,10 +254,16 @@ defmodule Fika.Parser do
     |> optional(function_ref_call)
     |> Helper.to_ast(:function_ref_call)
 
+  negation =
+    string("!")
+    |> parsec(:term)
+    |> Helper.to_ast(:negation)
+
   factor =
     choice([
       literal_exps,
-      non_literal_exps
+      non_literal_exps,
+      negation
```

Comment:
Can this go inside `non_literal_exps`right at the end?

Reply:
Also, we optimized the parser so a lot of things have been moved around. After you fetch the latest changes, please let me know if you need help understanding any change.